src/OpenFga.Sdk/Exceptions/ApiException.cs

```diff
@@ -54,20 +55,24 @@ protected ApiException(SerializationInfo serializationInfo, StreamingContext str
     /// <param name="request"><see cref="HttpRequestMessage"/></param>
     /// <param name="apiName"></param>
     /// <returns>An instance of a <see cref="ApiException"/> subclass containing the appropriate exception for this response.</returns>
-    public static async Task<ApiException> CreateSpecificExceptionAsync(HttpResponseMessage response, HttpRequestMessage request,
+    public static async Task<ApiException> CreateSpecificExceptionAsync(HttpResponseMessage? response, HttpRequestMessage request,
         string? apiName = null) {
-        switch ((int)response.StatusCode) {
-            case 401:
-            case 403:
+        var statusCode = response?.StatusCode;
+        switch (statusCode) {
+            case HttpStatusCode.Unauthorized:
+            case HttpStatusCode.Forbidden:
                 return await FgaApiAuthenticationError.CreateAsync(response, request, apiName).ConfigureAwait(false);
-            case 400:
-            case 442:
+            case HttpStatusCode.BadRequest:
+            case HttpStatusCode.UnprocessableEntity:
                 return await FgaApiValidationError.CreateAsync(response, request, apiName).ConfigureAwait(false);
-            case 429:
+            case HttpStatusCode.NotFound:
+                return await FgaApiNotFoundError.CreateAsync(response, request, apiName).ConfigureAwait(false);
+            case HttpStatusCode.TooManyRequests:
                 return await FgaApiRateLimitExceededError.CreateAsync(response, request, apiName).ConfigureAwait(false);
-            case 500:
-                return await FgaApiInternalError.CreateAsync(response, request, apiName).ConfigureAwait(false);
             default:
+                if (statusCode >= HttpStatusCode.InternalServerError && statusCode != HttpStatusCode.NotImplemented) {
```

Comment:
by definition, all >= 500 must be != 304, right?

Reply:
Yes, but `NotImplemented` is `501` not `304`